src/scenic/syntax/ast.py

```diff
@@ -202,3 +202,260 @@ def __init__(
+            "BackLeft",
+            "BackRight",
+        ],
+        target: ast.AST,
+        *args: any,
+        **kwargs: any
+    ) -> None:
+        super().__init__(*args, **kwargs)
+        self.position = position
+        self.target = target
+
+
+class Front(AST):
+    "Represents position of `front of` operator"
+
+    def __init__(self, *args: any, **kwargs: any) -> None:
+        super().__init__(*args, **kwargs)
+
+
+class Back(AST):
+    "Represents position of `back of` operator"
+
+    def __init__(self, *args: any, **kwargs: any) -> None:
+        super().__init__(*args, **kwargs)
+
+
+class Left(AST):
+    "Represents position of `left of` operator"
+
+    def __init__(self, *args: any, **kwargs: any) -> None:
+        super().__init__(*args, **kwargs)
+
+
+class Right(AST):
+    "Represents position of `right of` operator"
+
+    def __init__(self, *args: any, **kwargs: any) -> None:
+        super().__init__(*args, **kwargs)
+
+
+class FrontLeft(AST):
+    "Represents position of `front left of` operator"
+
+    def __init__(self, *args: any, **kwargs: any) -> None:
+        super().__init__(*args, **kwargs)
+
+
+class FrontRight(AST):
+    "Represents position of `front right of` operator"
+
+    def __init__(self, *args: any, **kwargs: any) -> None:
+        super().__init__(*args, **kwargs)
+
+
+class BackLeft(AST):
+    "Represents position of `back left of` operator"
+
+    def __init__(self, *args: any, **kwargs: any) -> None:
+        super().__init__(*args, **kwargs)
+
+
+class BackRight(AST):
+    "Represents position of `back right of` operator"
+
+    def __init__(self, *args: any, **kwargs: any) -> None:
+        super().__init__(*args, **kwargs)
+
+
+class DegOp(AST):
+    __match_args__ = ("operand",)
+
+    def __init__(self, operand: ast.AST, *args: any, **kwargs: any) -> None:
+        super().__init__(*args, **kwargs)
+        self.operand = operand
+
+
+class VectorOp(AST):
+    __match_args__ = ("left", "right")
+
+    def __init__(
+        self, left: ast.AST, right: ast.AST, *args: any, **kwargs: any
+    ) -> None:
+        super().__init__(*args, **kwargs)
+        self.left = left
+        self.right = right
+        self._fields = ["left", "right"]
+
+
+class FieldAtOp(AST):
+    __match_args__ = ("left", "right")
+
+    def __init__(
+        self, left: ast.AST, right: ast.AST, *args: any, **kwargs: any
+    ) -> None:
+        super().__init__(*args, **kwargs)
+        self.left = left
+        self.right = right
+
+
+class RelativeToOp(AST):
+    __match_args__ = ("left", "right")
+
+    def __init__(
+        self, left: ast.AST, right: ast.AST, *args: any, **kwargs: any
+    ) -> None:
+        super().__init__(*args, **kwargs)
+        self.left = left
+        self.right = right
+
+
+class OffsetAlongOp(AST):
+    __match_args__ = ("left", "middle", "right")
+
+    def __init__(
+        self, left: ast.AST, middle: ast.AST, right: ast.AST, *args: any, **kwargs: any
```

Comment:
This node represents `X offset along H by Y`. I wanted to give meaningful names to the parameters but I couldn't come up with good ones. Any ideas?

Reply:
`base`, `direction`, and `offset`?